packages/c/catch2/xmake.lua

```diff
@@ -40,6 +40,9 @@ package("catch2")
 
     on_component("main", function (package, component)
         component:add("links", "Catch2Main")
+        if package:is_plat("windows") then
+            component:add("ldflags", "-subsystem:console")
```

Comment:
why would catch2 force a console? people may want to use it in a GUI app and redirect stdout to widgets or a file

Reply:
The "main" component contains a `main` function, which is expected to be linked into a console app. For those who want a custom entry point they should only link to the `lib` component. See https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#cmake-targets